pinot-core/src/main/java/com/linkedin/pinot/core/operator/docvalsets/MultiValueDictIdSet.java

```diff
@@ -18,28 +18,30 @@
 import com.linkedin.pinot.common.data.FieldSpec.DataType;
 import com.linkedin.pinot.core.common.BaseBlockValSet;
 import com.linkedin.pinot.core.common.BlockValIterator;
+import com.linkedin.pinot.core.io.reader.ReaderContext;
 import com.linkedin.pinot.core.io.reader.SingleColumnMultiValueReader;
-import com.linkedin.pinot.core.operator.docvaliterators.MultiValueIterator;
-import com.linkedin.pinot.core.segment.index.ColumnMetadata;
+import com.linkedin.pinot.core.operator.docvaliterators.MultiValueDictIdIterator;
 
 
-public final class MultiValueSet extends BaseBlockValSet {
-  private ColumnMetadata columnMetadata;
-  private SingleColumnMultiValueReader mVReader;
+public final class MultiValueDictIdSet extends BaseBlockValSet {
+  private final SingleColumnMultiValueReader<? super ReaderContext> _reader;
+  private final int _numDocs;
+  private final DataType _dataType;
 
-  public MultiValueSet(SingleColumnMultiValueReader mVReader, ColumnMetadata columnMetadata) {
-    super();
-    this.mVReader = mVReader;
-    this.columnMetadata = columnMetadata;
+  public MultiValueDictIdSet(SingleColumnMultiValueReader<? super ReaderContext> reader, int numDocs,
```

Comment:
I think numDocs should be fetched from the reader, instead of being passed here.